### ISSUE TITLE
AWS resources are queried independent of the specified region

### ISSUE THREAD

**Issue by czunker** [mondoohq/cnquery#913]
**Describe the bug**
When querying AWS resource without a region, automatically us-east-1 is used. But still the queries show also other regions. 
Also when a region is specified with `AWS_REGION` env variable, I see resources from other regions.

I'm not sure whether this is a bug or a feature. At least the behavior is different form the AWS console and the message `→ no AWS region found, using us-east-1` a bit misleading.

When the region is only used for authentication, we should make this clear.

Here are some examples:
https://github.com/mondoohq/cnquery/issues/912

https://github.com/mondoohq/cnquery/pull/883#issuecomment-1423846556

https://github.com/mondoohq/cnquery/pull/883#issuecomment-1423877063

For s3, I'm not sure. It is a global service, but still a bucket has a region specified:
![image](https://user-images.githubusercontent.com/827818/217776394-9d0a4804-96f1-42fb-aa75-a659ffffa33b.png)

**To Reproduce**

1. Specify a region
2. Query AWS resources
3. See resources from other regions

**Expected behavior**
I would only expect resources form the specified region.

**Comment by vjeffrey** [comment]
ya,
there is a --region flag for filtering

we can improve the message here, the region is assigned is for authentication, because the api requires a region be set

but i can see how that message is confusing

**Comment by chris-rock** [comment]
Similar to other environment variables, we can read AWS_REGION in the aws cli and set the region value. This should happen in the CLI and not in the inventory.

**Comment by vjeffrey** [comment]
yes, we do read that value, that's what's used for the api. the --option région flag is what is used for the filtering.
do people feel like we should be region filtering based on the value of the aws region env var?

**Comment by mariuskimmina** [comment]
Personally I think the behavior should be as close to the aws-cli as possible, since that is what people are used to (and we are using the same env var). `AWS_REGION` and `--region` should both limit the results to only that region. I don't think people expect to get multi-region results.

**Comment by tas50** [comment]
Reviewed as part of a full backlog triage. This appears resolved in the current codebase:

> providers/aws/connection/filters.go:30-35 parses 'regions'/'exclude:regions' into GeneralDiscoveryFilters; connection.go:103 wires DiscoveryFiltersFromOpts from conf.Discover.GetFilter(); connection.go:341-347 Regions() returns regionLimits (h.Filters.General.Regions) when set, short-circuiting enumeration, and lines 377-384 apply ExcludeRegions.

Core ask (restrict queries to specified regions) is satisfied via discovery filters; the secondary 'us-east-1 misleading message' note in the issue is not addressed but is incidental.

Closing as fixed. If you still see this on a recent release, please reopen with the version.
